src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.impl.response.OrderResponseMapper;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.request.OrderRequestDto;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final OrderResponseMapper orderResponseMapper;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+
+    public OrderController(OrderService orderService,
+                           OrderResponseMapper orderResponseMapper,
+                           UserService userService,
+                           ShoppingCartService shoppingCartService) {
+        this.orderService = orderService;
+        this.orderResponseMapper = orderResponseMapper;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getAllOrdersByUserId(@RequestParam Long userId) {
+        User user = userService.get(userId);
+        return orderService
+                .getOrdersHistory(user)
+                .stream()
+                .map(orderResponseMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto complete(@RequestBody OrderRequestDto orderRequestDto) {
```

Comment:
![image](https://user-images.githubusercontent.com/117590814/219952088-54d80a23-8b73-4268-8010-4d6d7b522f4b.png)
the point here is to complete the order that user made